lib/resources/resourceBase.js

```diff
@@ -64,7 +64,10 @@ var ResourceBase = function (endpoint, config) {
         }
 
         if (body && body.error) {
-          return reject(body.error);
+          var error = new Error(body.error.message + ' (' + body.error.status_code + ')');
+          error.statusCode = body.error.status_code;
+          error.statusMessage = body.error.message;
```

Comment:
Snake case because, right now, for people handling normal errors, they can be checking `err.message` and `err.status_code` because that's how `body.error` is structured. By making it snake, it won't break people's implementation for normal errors. Neither of these properties (snake or camel) are normal props for Error objects so it's not like it's going against "normal Error conventions", though it could be considered going against JS conventions. But yeah, that was my reasoning for it, though I can see the argument for camel. What do you think?